pkg/services/sqlstore/dashboard_snapshot.go

```diff
@@ -45,22 +49,32 @@ func CreateDashboardSnapshot(cmd *models.CreateDashboardSnapshotCommand) error {
 			expires = time.Now().Add(time.Second * time.Duration(cmd.Expires))
 		}
 
-		snapshot := &models.DashboardSnapshot{
-			Name:              cmd.Name,
-			Key:               cmd.Key,
-			DeleteKey:         cmd.DeleteKey,
-			OrgId:             cmd.OrgId,
-			UserId:            cmd.UserId,
-			External:          cmd.External,
-			ExternalUrl:       cmd.ExternalUrl,
-			ExternalDeleteUrl: cmd.ExternalDeleteUrl,
-			Dashboard:         cmd.Dashboard,
-			Expires:           expires,
-			Created:           time.Now(),
-			Updated:           time.Now(),
+		encode, err := cmd.Dashboard.Encode()
```

Comment:
As I was making these changes it made me think of https://github.com/grafana/grafana/issues/26836 as we had been discussing it only recently. I'm not certain yet but after a brief look I think this will be base64 encoded and since we're already now storing it as a `blob`, we may be good 😄

Reply:
Thought I added a comment here. Well, here it goes again. Happened to notice the other day that we do encrypt and encode for OAuth tokens that we store in the database. But maybe that's due to the database type DB_Text used?

https://github.com/grafana/grafana/blob/90303b5a03215f5dbcd3215a182f3251e0f0a53c/pkg/services/sqlstore/user_auth.go#L275-L283

Reply:
Hmm interesting.. Maybe it's better to keep them consistent then. Will explore that option and see how it turns out 😄

Reply:
Are we solving #26836 with this PR?

Reply:
No not directly. In this case we're just focussing on the dashboards within `dashboard _snapshot`s, but now is a good opportunity to do the right thing and follow up later with the fix for #26836

Reply:
@wbrowne Ah... That difference was lost on me.